prowler/providers/m365/services/entra/entra_service.py

```diff
@@ -385,6 +398,11 @@ async def fetch_role_members(directory_role):
                         True if (user.on_premises_sync_enabled) else False
                     ),
                     directory_roles_ids=user_roles_map.get(user.id, []),
+                    is_mfa_capable=(
+                        registration_details.get(user.id, {}).is_mfa_capable
+                        if registration_details.get(user.id, {}) is not None
+                        else False
+                    ),
```

Comment:
```suggestion
                    is_mfa_capable=(
                        registration_details.get(user.id, {}).is_mfa_capable
                        if user.id in registration_details 
                        else False
                    ),
```

`{}` is not None, so if the user is not on `registration_details` (not sure if that's possible) it would return `{}` and `.is_mfa_capable` would fail.

Reply:
I believe that information is always available.
Once said that, setting it to `None` instead of `{}` should produce the desired result. If we can’t retrieve `registration_details`, we set it to `None`, and as a result, `is_mfa_capable` ends up being set to `False` (non compliant).